packages/ec-core/src/individual/ec.rs

```diff
@@ -93,16 +96,28 @@ impl<GG, S> IndividualGenerator<GG, S> {
 
 /// A trait for adding a scorer to a genome generator, creating
 /// an `IndividualGenerator`.
-pub trait WithScorer<Scorer> {
+pub trait WithScorer {
     /// Add a scorer to the genome generator, creating an `IndividualGenerator`.
-    fn with_scorer(self, scorer: Scorer) -> IndividualGenerator<Self, Scorer>
+    fn with_scorer<S, G, R>(self, scorer: S) -> IndividualGenerator<Self, S>
+    where
+        Self: Sized,
+        S: Scorer<G, R>;
+
+    fn with_scorer_fn<F, G, R>(self, f: F) -> IndividualGenerator<Self, FnScorer<F>>
     where
-        Self: Sized;
+        Self: Sized,
+        F: Fn(&G) -> R,
```

Comment:
```rs
fn with_scorer_fn<F, G>(self, f: F) -> IndividualGenerator<Self, FnScorer<F>>
    where
        Self: Sized,
        FnScorer<F>: Scorer<G>,
```

Reply:
We chose to not go this route because (as esitsu@Twitch pointed out) if we replace `F: Fn(&G) -> R` with `FnScorer<F>: Scorer<G>`, someone reading the code doesn't know what they can pass in as `f` without going off and reading `FnScorer` as well. So it would have nicely removed the `R`, but the mental cost of that indirection seemed to outweigh that benefit.

Reply:
:+1: